extensions/amp-recaptcha-input/0.1/amp-recaptcha-service.js

```diff
@@ -193,6 +207,80 @@ export class AmpRecaptchaService {
     }
   }
 
+  /**
+   * Function to create our bootstrap iframe.
+   *
+   * @param {string} sitekey
+   * @return {!Promise<!Element>}
+   * @private
+   */
+  createRecaptchaFrame_(sitekey) {
+
+    const iframe = this.win_.document.createElement('iframe');
+
+    return this.getRecaptchaFrameSrc_().then(recaptchaFrameSrc => {
+      iframe.src = recaptchaFrameSrc;
+      iframe.ampLocation = parseUrlDeprecated(this.win_.location.href);
+      iframe.setAttribute('scrolling', 'no');
+      iframe.setAttribute('data-amp-3p-sentinel', 'amp-recaptcha');
+      iframe.setAttribute('name', JSON.stringify(dict({
+        'sitekey': sitekey,
+        'sentinel': 'amp-recaptcha',
+      })));
+      setStyle(iframe, 'border', 'none');
+      /** @this {!Element} */
+      iframe.onload = function() {
+        // Chrome does not reflect the iframe readystate.
+        this.readyState = 'complete';
```

Comment:
what's the readyState here for?

Reply:
This must be done for the `loadPromise()`, it is something implemented by all AMP elements, and I can find it in the source if you would like 😄